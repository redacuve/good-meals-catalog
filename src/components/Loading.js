import React from 'react';

function Loading() {
  return (
    <div className="w-full bg-white opacity-75">
      <span className="text-indigo-500 my-6 mx-auto block w-0">
        <i className="fas fa-spinner fa-pulse fa-5x" />
      </span>
    </div>
  );
}

export default Loading;
