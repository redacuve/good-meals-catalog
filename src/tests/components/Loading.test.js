import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Loading from '../../components/Loading';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Loading />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('render Loading icon correctly', () => {
  const { getByTestId } = render(<Loading />);
  expect(getByTestId('icon-container')).toContainHTML(
    '<i class="fas fa-spinner fa-pulse fa-5x"></i>',
  );
});

it('render container correctly', () => {
  const { getByTestId } = render(<Loading />);
  const child = getByTestId('icon-container');
  expect(getByTestId('loading-container')).toContainElement(child);
});

it('matches snapshot 1', () => {
  const tree = renderer.create(<Loading />).toJSON();
  expect(tree).toMatchSnapshot();
});
