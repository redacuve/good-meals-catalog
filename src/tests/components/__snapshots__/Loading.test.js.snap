// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="w-full bg-white opacity-75"
  data-testid="loading-container"
>
  <span
    className="text-indigo-500 my-6 mx-auto block w-0"
    data-testid="icon-container"
  >
    <i
      className="fas fa-spinner fa-pulse fa-5x"
    />
  </span>
</div>
`;
